bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/AbstractDiscoveryService.java

```diff
@@ -249,6 +258,20 @@ protected synchronized void stopScan() {
      *            Holds the information needed to identify the discovered device.
      */
     protected void thingDiscovered(DiscoveryResult discoveryResult) {
+        if (this.i18nProvider != null && this.locale != null) {
```

Comment:
Just an idea: We could change this method signature to

```Java
protected void thingDiscovered(DiscoveryResult discoveryResult, final @Nullable TranslationProvider translationProvider, final @Nullable Locale locale)
```

and create an obsolete method

```Java
protected final void thingDiscovered(DiscoveryResult discoveryResult) {
    thingDiscovered(discoveryResult, null, null);
}
```

Why at all? To ensure that the "I read not every updated documentation" developer knows that he should provide the two additional references to the abstract base class implementation.

Reply:
The good thing is that the developers do not need to provide these two additional references at all - only the ones who want to provide translated labels would have to make sure that their DiscoveryService gets LocaleProvider and TranslationProvider injected.